src/Bindings/LuaState.h

```diff
@@ -112,6 +115,63 @@ class cLuaState
 	} ;
 	
 	
+	/** Represents a callback to Lua that C++ code can call.
+	Is thread-safe and unload-safe.
+	When the Lua state is unloaded, the callback returns an error instead of calling into non-existent code.
+	To receive the callback instance from the Lua side, use RefStack() or (better) cLuaState::GetStackValue(). */
+	class cCallback
+	{
+	public:
+		/** Creates an unbound callback instance. */
+		cCallback(void) = default;
+
+		~cCallback()
+		{
+			Clear();
+		}
+
+		/** Calls the Lua callback, if still available.
+		Returns true if callback has been called.
+		Returns false if the Lua state isn't valid anymore. */
+		template <typename... Args>
+		bool Call(Args &&... args)
```

Comment:
The lambda object should be valid for as long as the std::function object is, because its a value type. If its the members that have insufficient lifetimes, then I don't see what lambdas have to do with it, you get exactly the same problems from manual functor classes. 
Also I don't see any references to cCallbackContainer in the PR, so I don't really understand the second point.